core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1710,6 +1695,18 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /**
+   * Get the segments in the log which have a non-empty offset intersection with the
+   * inbound swap segment
+   */
+  private def swapSegmentIntersection(swapSegment: LogSegment): Iterable[LogSegment] = {
+    lock synchronized {
```

Comment:
Why is this synchronized whereas the previously inline code was not?

Reply:
I guess we don't need the lock during segment loading. I added this initially as a separate method following the pattern of `logSegments` because I thought it might be a more general method. But since we only have one usage and the code is small enough, I think I will just inline it.